src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp

```diff
@@ -318,6 +285,145 @@ class boss_ichoron : public CreatureScript
             if (victim->GetTypeId() == TYPEID_PLAYER)
                 Talk(SAY_SLAY);
         }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            if (!bIsFrenzy && HealthBelowPct(25) && !bIsExploded)
+            {
+                Talk(SAY_ENRAGE);
+                DoCast(me, SPELL_FRENZY, true);
+                bIsFrenzy = true;
+            }
+
+            if (!bIsFrenzy)
+            {
+                if (!bIsExploded)
+                {
+                    if (!me->HasAura(SPELL_PROTECTIVE_BUBBLE))
+                    {
+                        bIsExploded = true;
+                        Talk(SAY_SHATTER);
+                        DoCast(SPELL_BURST);
+                        me->RemoveAllAuras();
+                        burstTimer = 15000;
+
+                        std::list<Creature*> summonTargets;
+                        GetCreatureListWithEntryInGrid(summonTargets, me, NPC_ICHORON_SUMMON_TARGET, 200.0f);
+                        std::list<Creature*>::iterator itr = summonTargets.begin();
+
+                        for (uint8 i = 0; i < 10; i++)
+                        {
+                            std::advance(itr, urand(0, summonTargets.size() - 1)); // I take a random minion in the list
+                            Position targetPos = (*itr)->GetRandomNearPosition(10.0f);
+                            itr = summonTargets.begin();
+                            TempSummon* globule = me->SummonCreature(NPC_ICHOR_GLOBULE, targetPos, TEMPSUMMON_CORPSE_DESPAWN);
+                            DoCast(globule, SPELL_WATER_GLOBULE_VISUAL);
+
+                            float minDistance = 1000.0f;
+                            uint8 nextPath = 0;
+                            // I move the globules to next position. the 10 positions are in couples, defined in globulePaths, so i have to increase by 2.
+                            for (uint8 gpath = 0; gpath < 10; gpath += 2) 
+                            {
+                                if (globule->GetDistance(globulePaths[gpath]) < minDistance)
+                                {
+                                    minDistance = globule->GetDistance(globulePaths[gpath]);
+                                    nextPath = gpath;
+                                }
+                            }
+
+                            globule->GetAI()->SetData(DATA_GLOBULE_PATH, nextPath);
+                        }
+                        return;
+                    }
+
+                    if (me->HasUnitState(UNIT_STATE_CASTING))
+                        return;
+
+                    events.Update(diff);
+
+                    switch (uint32 eventId = events.ExecuteEvent())
```

Comment:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp: In member function ‘virtual void boss_ichoron::boss_ichoronAI::UpdateAI(uint32)’:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp:348:36: warning: unused variable ‘eventId’ [-Wunused-variable]
                     switch (uint32 eventId = events.ExecuteEvent())
                                    ^